frontend/app/services/personal-information-service.server.ts

```diff
@@ -145,23 +152,26 @@ function createPersonalInformationService() {
   }
 
   function toPersonalInformation(personalInformationApiResponse: PersonalInformationApiResponse): PersonalInfo {
-    const addressList = personalInformationApiResponse.Client.PersonContactInformation?.Address;
+    const addressList = personalInformationApiResponse?.Client.PersonContactInformation?.Address;
     const homeAddressList = addressList?.filter((address) => address.AddressCategoryCode.ReferenceDataName == 'Home');
     const mailingAddressList = addressList?.filter((address) => address.AddressCategoryCode.ReferenceDataName == 'Mailing');
 
     return {
-      applictantId: personalInformationApiResponse.Client.ClientIdentification ? personalInformationApiResponse.Client.ClientIdentification.filter((clientInfoDto) => clientInfoDto.IdentificationCategoryText === 'Applicant ID').at(0)?.IdentificationID : '',
-      clientId: personalInformationApiResponse.Client.ClientIdentification ? personalInformationApiResponse.Client.ClientIdentification.filter((clientInfoDto) => clientInfoDto.IdentificationCategoryText === 'Client Number').at(0)?.IdentificationID : '',
-      firstName: personalInformationApiResponse.Client.PersonName ? personalInformationApiResponse.Client.PersonName.at(0)?.PersonGivenName?.at(0) : '',
-      lastName: personalInformationApiResponse.Client.PersonName ? personalInformationApiResponse.Client.PersonName.at(0)?.PersonSurName?.at(0) : '',
-      emailAddress: personalInformationApiResponse.Client.PersonContactInformation ? personalInformationApiResponse.Client.PersonContactInformation.EmailAddress.EmailAddressID : '',
-      phoneNumber: personalInformationApiResponse.Client.PersonContactInformation ? personalInformationApiResponse.Client.PersonContactInformation.TelephoneNumber.FullTelephoneNumber : '',
+      applictantId: personalInformationApiResponse?.Client.ClientIdentification
+        ? personalInformationApiResponse.Client.ClientIdentification.filter((clientInfoDto) => clientInfoDto.IdentificationCategoryText === 'Applicant ID').at(0)?.IdentificationID
+        : '',
+      clientId: personalInformationApiResponse?.Client.ClientIdentification ? personalInformationApiResponse.Client.ClientIdentification.filter((clientInfoDto) => clientInfoDto.IdentificationCategoryText === 'Client Number').at(0)?.IdentificationID : '',
+      firstName: personalInformationApiResponse?.Client.PersonName ? personalInformationApiResponse.Client.PersonName.at(0)?.PersonGivenName?.at(0) : '',
+      lastName: personalInformationApiResponse?.Client.PersonName ? personalInformationApiResponse.Client.PersonName.at(0)?.PersonSurName?.at(0) : '',
+      emailAddress: personalInformationApiResponse?.Client.PersonContactInformation ? personalInformationApiResponse.Client.PersonContactInformation.EmailAddress.EmailAddressID : '',
+      phoneNumber: personalInformationApiResponse?.Client.PersonContactInformation ? personalInformationApiResponse.Client.PersonContactInformation.TelephoneNumber.FullTelephoneNumber : '',
```

Comment:
Is it possible to remove the defaulted empty values for each of these fields and leave them as `undefined` instead? Our `personalInfoDtoSchema` indicates that these fields are optional.

We should be able to use optional chaining and do something like:

```suggestion
      phoneNumber: personalInformationApiResponse?.Client.PersonContactInformation?.TelephoneNumber.FullTelephoneNumber,
```

Reply:
Removed: 102180defa666472647423336ecff5b8c2e4f0c2